lib/associations/base.js

```diff
@@ -1,6 +1,13 @@
 'use strict';
+const AssociationError = require('./../errors').AssociationError;
 
-var Association = function() {};
+var Association = function(source, options) {
```

Comment:
If we introduce a constructor here, it should take `source, target, options` like the others, and assign those so the subclasses don't have to